nano/node/transport/tcp.hpp

```diff
@@ -55,7 +55,7 @@ namespace transport
 	class tcp_server;
 	class tcp_channels;
 
-	class channel_tcp : public nano::transport::channel
+	class channel_tcp : public nano::transport::channel, public std::enable_shared_from_this<channel_tcp>
```

Comment:
Maybe enable_shared_from_this should be on the base class nano::transport::channel?

Reply:
I'm not aware of an easy way to get shared_from_this working with a base class, is there any? From what I understand a form of CRTP pattern will be needed and that's not going to be easier to implement than inheriting from shared_from_this and a virtual `shared ()` function to make it polymorphic.